### PR TITLE
Use libc constants in sys/signal.rs.

### PR DESCRIPTION
**Pull request by fiveop**
Work toward #264.

### CODE REVIEW

**Comment by kamalmarhubi** [comment]
`+45 −277`

I like the sound of that diffstat!

File: src/sys/signal.rs [hunk 1]
```diff
@@ -44,298 +44,69 @@ pub const SIGEMT: libc::c_int = 7;
...
-    }
-
-    #[cfg(any(target_os = "openbsd", target_os = "netbsd"))]
-    #[repr(C)]
-    pub struct sigaction {
-        pub sa_handler: extern fn(libc::c_int),
-        pub sa_mask: sigset_t,
-        pub sa_flags: SockFlag,
+bitflags!{
+    flags HowFlag: libc::c_int {
```
Comment (kamalmarhubi) [inline src/sys/signal.rs]: should we rename this to `SigFlags` while here?

File: src/sys/signal.rs [hunk 2]
```diff
@@ -44,298 +44,69 @@ pub const SIGEMT: libc::c_int = 7;
...
-        pub sa_mask: sigset_t,
-        sa_restorer: *mut libc::c_void,
-        sa_resv: [libc::c_int; 1],
-    }
-
-    #[repr(C)]
-    pub struct sigset_t {
-        __val: [libc::c_ulong; 32],
+bitflags!{
+    flags SockFlag: libc::c_int {
```
Comment (kamalmarhubi) [inline src/sys/signal.rs]: should we rename this to SaFlags while we're here?

File: src/sys/signal.rs [hunk 3]
```diff
@@ -44,298 +44,69 @@ pub const SIGEMT: libc::c_int = 7;
...
-        pub fn sigfillset(set: *mut sigset_t) -> c_int;
-        pub fn sigismember(set: *const sigset_t, signum: c_int) -> c_int;
-
-        pub fn pthread_sigmask(how: c_int, set: *const sigset_t, oldset: *mut sigset_t) -> c_int;
-
-        pub fn kill(pid: pid_t, signum: c_int) -> c_int;
-        pub fn raise(signum: c_int) -> c_int;
-
-        pub fn sigwait(set: *const sigset_t, sig: *mut c_int) -> c_int;
+        pub fn sigwait(set: *const libc::sigset_t, sig: *mut libc::c_int) -> libc::c_int;
```
Comment (kamalmarhubi) [inline src/sys/signal.rs]: is the issue here that `sigwait` is missing in libc?

Comment (kamalmarhubi) [inline src/sys/signal.rs]: ah yes it is missing. opened https://github.com/rust-lang/libc/pull/203

Comment (kamalmarhubi) [inline src/sys/signal.rs]: missed your https://github.com/rust-lang/libc/pull/201 moving on

**Comment by kamalmarhubi** [comment]
Other than possible renaming of flags types, r=me. This is great!

File: src/sys/signal.rs (hunk 3 above)
Comment (fiveop) [inline src/sys/signal.rs]: I already opened a PR for that.

**Comment by fiveop** [comment]
@homu r=@kamalmarhubi

**Pull request merged**